Packet++/src/WireGuardLayer.cpp

```diff
@@ -0,0 +1,324 @@
+#define LOG_MODULE PacketLogModuleWireGuardLayer
+
+#include "UdpLayer.h"
+#include "IPv4Layer.h"
+#include "IPv6Layer.h"
+#include "WireGuardLayer.h"
+#include "Logger.h"
+#include <sstream>
+#include <iomanip>
+#include <cstring>
+
+namespace pcpp
+{
+	void WireGuardLayer::parseNextLayer()
+	{
+		size_t headerLen = getHeaderLen();
+		if (m_DataLen <= headerLen || headerLen == 0)
+			return;
+		m_NextLayer = WireGuardLayer::parseWireGuardLayer(m_Data, m_DataLen, this, m_Packet);
+	}
+
+	WireGuardLayer* WireGuardLayer::parseWireGuardLayer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet)
+	{
+		if (dataLen < sizeof(WireGuardLayer::wg_common_header))
+			return nullptr;
+		wg_common_header* wgHeader = reinterpret_cast<wg_common_header*>(data);
+
+		switch (wgHeader->messageType)
+		{
+		case static_cast<uint8_t>(WireGuardMessageType::HandshakeInitiation):
+			return new WireGuardHandshakeInitiationLayer(data, dataLen, prevLayer, packet);
+		case static_cast<uint8_t>(WireGuardMessageType::HandshakeResponse):
+			return new WireGuardHandshakeResponseLayer(data, dataLen, prevLayer, packet);
+		case static_cast<uint8_t>(WireGuardMessageType::CookieReply):
+			return new WireGuardCookieReplyLayer(data, dataLen, prevLayer, packet);
+		case static_cast<uint8_t>(WireGuardMessageType::TransportData):
+			return new WireGuardTransportDataLayer(data, dataLen, prevLayer, packet);
+		default:
+			return nullptr;
+		}
+	}
+
+	std::string WireGuardLayer::getMessageTypeAsString() const
+	{
+		// Assuming you have a method to retrieve the message type from the header
+		wg_common_header* wgHeader = reinterpret_cast<wg_common_header*>(this->getData());
+		switch (wgHeader->messageType)
```

Comment:
If we have a method like `getMessageType()` that returns `WireGuardMessageType`, maybe we should use it here?

Reply:
Sure, I've made the changes.